### ISSUE TITLE
Questions about Installation & Setup Scripts

### ISSUE THREAD

**Issue by JihoChoi** [OpenRobotLab/OV_PARTS#7]
## Questions about Installation & Setup Scripts

Thanks for sharing the research and code.
Could anyone help me set up the environment to run the evaluation script?

### Questions

1. There is no `third_party/CLIP` directory, which is stated in the Installation section of `README.md`.
    1. Is it the `open_clip`?
1. Detectron2 is not included in the installation instructions, do I need to separately download it?
    1. `python -m pip install 'git+https://github.com/facebookresearch/detectron2.git'`
1. After running the below scripts, I am getting the below error messages when running the evaluation script.
    1. Error: `No module named 'transformers.models.encoder_decoder'`
        1. -> `pip install transformers` (resolved ? - not sure if they are the same)
        2. the `transformers` directory in this repository doesn't contain `encoder_decoder` module
    1. Error: `No module named 'clip'`
        1. -> `pip install clip` (resolved ? - not sure if they are the same)
    1. Error `ImportError: tokenizers>=0.11.1,!=0.11.3,<0.14 is required for a normal functioning of this module, but found tokenizers==0.15.1.`
        1. Still getting the below errors.
        > ImportError: tokenizers>=0.11.1,!=0.11.3,<0.14 is required for a normal functioning of this module, but found tokenizers==0.15.1.
        > Try: pip install transformers -U or pip install -e '.[dev]' if you're working with git main

1. Could someone please share what needs to be done to configure the environment so that the model runs?

### My Current Setup Script

```bash
# ------------------------
#     1. Setup
# ------------------------
conda create -n ovparts python=3.8
conda activate ovparts
pip install -r requirements.txt

# original instruction in README
# cd third_party/CLIP  # no such directory
# python -m pip install -e .

# alternative try
cd open_clip
make install  # Reference: https://github.com/mlfoundations/open_clip
cd ..

# ------------------------
#     2. Install Detectron2
# ------------------------
# https://detectron2.readthedocs.io/en/latest/tutorials/install.html
python -m pip install 'git+https://github.com/facebookresearch/detectron2.git'
pip install pillow==9.5.0  # Due to `AttributeError: module 'PIL.Image' has no attribute 'LINEAR'` error

# ------------------------
#     3. Prepare Datasets
# ------------------------
# Pascal-Part-116 - https://drive.google.com/file/d/1f5kqrM2_iK_bWmQBW3rdSnGrnke4PUbX/view

pip install gdown
mkdir Datasets
cd Datasets
gdown https://drive.google.com/uc?id=1f5kqrM2_iK_bWmQBW3rdSnGrnke4PUbX
tar -xzf PascalPart116.tar.gz
cd ..

pip install fire
python baselines/data/datasets/mask_cls_collect.py Datasets/PascalPart116/annotations_detectron2_part/val Datasets/PascalPart116/annotations_detectron2_part/val_part_label_count.json


# ------------------------
#     4. Prepare Weights
# ------------------------
mkdir weights
cd weights
gdown https://drive.google.com/uc?id=1WkCu3-KA2Oho5xzBXDR_HUmBvvKKYSQM
cd ..


# ------------------------
#     5. Eval
# ------------------------
python train_net.py \
    --num-gpus 8 \
    --config-file configs/zero_shot/clipseg_voc.yaml \
    --eval-only MODEL.WEIGHTS ./weights/clipseg_ft_VA_L_F_voc.pth


```

**Comment by kellyiss** [comment]
Sorry for the environment conflict about clip. Both clip and openclip are used in the baselines.
Please check the new updates for resolving the conflict and follow the updated README.md to try again!
No need to install clip now.
And yes, detectron2 should be installed.

**Comment by JihoChoi** [comment]
Thank you very much for the comment.
However, the updated [requirements.txt] and [README.md] don't work as well. ([074f87b](https://github.com/OpenRobotLab/OV_PARTS/commits/main/))
Could you share with us how you are able to run the model after initializing the new conda env (`"conda create -n ovparts2 python=3.8 -y"`)?
It would be great if you could share with us what scripts you are able to run for the setup manuals.
I tried a variety of versions torch versions with docker and conda envs.

Thank you very much.

![스크린샷 2024-02-07 오전 11 19 28](https://github.com/OpenRobotLab/OV_PARTS/assets/5732211/686a75cc-5656-492f-807f-44a63d2e8163)

**Comment by kellyiss** [comment]
Can you try pip install torch==1.12.1+cu113 torchvision==0.13.1+cu113 --extra-index-url https://download.pytorch.org/whl/cu113 ?

**Comment by kellyiss** [comment]
torch==1.12.1+cu113 means I installed torch version 1.12.1 with CUDA 11.3. Please make sure you install torch with your CUDA version.

**Comment by JihoChoi** [comment]
Thank you very much for your help.

I matched the environment of my docker container to `CUDA 11.3`
However, as stated in `README.md`, I used python 3.8.
I believe conda python 3.8 (pip 3.8), doesn't support `torch==1.12.1+cu113` and producing `ERROR: No matching distribution found for torch==1.12.1+cu113`.

Thank you for your help.
`pip install torch==1.12.1+cu113 torchvision==0.13.1+cu113 --extra-index-url https://download.pytorch.org/whl/cu113` resolves the issue.

However, now below message produce when running the `pip install -r requirements.txt`

> ERROR: Ignored the following versions that require a different python version: 0.22.0 Requires-Python >=3.9; 0.22.0rc1 Requires-Python >=3.9; 1.4.0 Requires-Python >=3.9; 1.4.0rc1 Requires-Python >=3.9
ERROR: Could not find a version that satisfies the requirement detectron2 (from versions: none)
ERROR: No matching distribution found for detectron2

![image](https://github.com/OpenRobotLab/OV_PARTS/assets/5732211/4724383e-33d9-4d5c-846c-16b01c8090ad)

Thank you for your help.
